test.py

```diff
@@ -89,6 +89,37 @@ def test_out_of_bounds(self):
             for i in range(100):
                 stack.push(str(i))
                 self.assertFalse(stack.is_empty())
+    
+    def test_push_pop(self): # testing pushing and popping 
```

Comment:
This one is kinda already covered by the general case. Maybe try to test some failure cases?